src/Controller/BaseController.php

```diff
@@ -68,4 +68,18 @@ public function getPreviousUrl(Request $request)
         $redirect = $request->headers->get('referer') ?? $this->generateUrl('home');
         return $redirect;
     }
+
+    public function restrictViewAccess($entity)
+    {
+        if ($entity->getIsPrivate() && $entity->getUser() != $this->getUser()) {
+            throw $this->createAccessDeniedException();
+        }
```

Comment:
@daria-sieroshtan 
that is a very good place to use an interface, eg `Blameablenterface`.
I bet your IDE also thinks so